server/plugin/client.go

```diff
@@ -154,12 +154,43 @@ func (c *client) CreateSubscription(body *serializers.CreateSubscriptionRequestP
 	case constants.ServiceTypeBoards:
 		payload.PublisherInputs = serializers.PublisherInputsBoards{
 			ProjectID: project.ProjectID,
+			AreaPath:  body.AreaPath,
 		}
 	case constants.ServiceTypeRepos:
-		payload.PublisherInputs = serializers.PublisherInputsRepos{
-			ProjectID:  project.ProjectID,
-			Repository: body.Repository,
-			Branch:     body.TargetBranch,
+		switch body.EventType {
+		case constants.SubscriptionEventCodePushed:
+			payload.PublisherInputs = serializers.PublisherInputsRepos{
+				ProjectID:  project.ProjectID,
+				Repository: body.Repository,
+				Branch:     body.TargetBranch,
+				PushedBy:   body.PushedBy,
+			}
+		case constants.SubscriptionEventPullRequestMerged:
+			payload.PublisherInputs = serializers.PublisherInputsRepos{
+				ProjectID:                    project.ProjectID,
+				Repository:                   body.Repository,
+				Branch:                       body.TargetBranch,
+				MergeResult:                  body.MergeResult,
+				PullRequestCreatedBy:         body.PullRequestCreatedBy,
+				PullRequestReviewersContains: body.PullRequestReviewersContains,
+			}
+		case constants.SubscriptionEventPullRequestUpdated:
+			payload.PublisherInputs = serializers.PublisherInputsRepos{
+				ProjectID:                    project.ProjectID,
+				Repository:                   body.Repository,
+				Branch:                       body.TargetBranch,
```

Comment:
You should add the common fields above the switch case, and then inside the switch , you can add fields according to the conditions.